node_io/lib/src/directory.dart

```diff
@@ -65,23 +67,28 @@ class Directory extends FileSystemEntity implements io.Directory {
   }
 
   @override
-  Stream<io.FileSystemEntity> list(
+  Stream<FileSystemEntity> list(
       {bool recursive: false, bool followLinks: true}) {
+    if (recursive)
+      throw new UnsupportedError('Recursive list is not supported in Node.');
     final controller = new StreamController<FileSystemEntity>();
 
-    void callback(err, files) {
+    void callback(err, [files]) {
       if (err != null) {
         controller.addError(err);
         controller.close();
       } else {
         for (var filePath in files) {
+          // Need to append the original path to build a proper path
+          filePath = join(path, filePath);
           final stat = FileStat.statSync(filePath);
           if (stat.type == io.FileSystemEntityType.file) {
             controller.add(new File(filePath));
           } else if (stat.type == io.FileSystemEntityType.directory) {
             controller.add(new Directory(filePath));
           } else {
-            throw new UnimplementedError('Link entities not implemented yet.');
+            // Yes not supported but no throw yet so it works for other cases
+            // throw new UnimplementedError('Link entities not implemented yet.');
```

Comment:
Could you add a TODO here as well? Would be easier to find later.

Reply:
ok